### PR TITLE
Allow to use third-party S3-compatible storages

### LINKED ISSUE #3: Allow to use third-party S3-compatible storages
Hello, i'm have a monkey-patch for this provider to use third-party S3-compatible storages. Will prepare PR for this.

### PR DESCRIPTION
**Pull request by shyr1punk**
fixes #3

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -23,6 +23,7 @@ module.exports = {
     AWS.config.update(awsConfig);
 
     const S3 = new AWS.S3({
+      endpoint: config.endpoint,
```
Comment (alexmukho) [inline lib/index.js:26]: According to documentation, endpoint should be an instance of AWS.Endpoint:
https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/S3.html#endpoint-property
Does it work fine with plain string?

I assume AWS uses the default endpoint if config.endpoint is null and there are no issues with that?

Comment (shyr1punk) [reply lib/index.js:26]: Actually it is:
```
(property) ServiceConfigurationOptions.endpoint?: string | AWS.Endpoint
The endpoint URI to send requests to. The default endpoint is built from the configured region. The endpoint should be a string like 'https://{service}.{region}.amazonaws.com' or an Endpoint object.
```

And i'm using `http://sber***.s***1.***cloud.ru`

Comment (shyr1punk) [reply lib/index.js:26]: https://github.com/aws/aws-sdk-js/blob/master/lib/service.d.ts#L74

Comment (alexmukho) [reply lib/index.js:26]: Ok, I see.

**Review by alexmukho — APPROVED** [review]

**Pull request merged**